frontend/src/components/Tabs/DoctorTabs/ContactPatients.js

```diff
@@ -0,0 +1,168 @@
+            });
+        }
+    })
+
+    useEffect(async () => {
+        try {
+            const res = await axios.get(`/doctor/${currentUser.user.uid}/patientArray`);
+            console.log('Patients: ', res.data.data);
+            setPatients([...res.data.data]);
+        } catch (error) {
+            console.log('Error fetching list of patients: ', error);
+        }
+    }, [])
+
+    const fetchChatMessages = async (patientIndex) => {
+        try {
+            const response = await axios.get(`/user/chats/${patients[patientIndex].uid}/${currentUser.user.uid}`);
+            console.log('Received patient chats: ', response.data);
+            setChats([...response.data]);
+            setSelectedChatIndex(patientIndex);
+        } catch (error) {
+            console.log('Error fetching chat messages: ', error);
+        }
+    }
+
+    const sendMessage = () => {
+        const patientId = patients[selectedChatIndex].uid;
+        const doctorId = currentUser.user.uid;
+        const chatId = patientId + '_' + doctorId;
+        const msg = {chatId: chatId, message: messageInput, senderId: doctorId, receiverId: patientId, priority: isPriority};
+        console.log('Is socket connected: ', socket.connected);
+        socket.emit('private-message', msg);
+        setChats([...chats, msg]);
+        setMessageInput('');
+    }
+
+    patients.map((patient, index) => {
+        return <Tab.Content onClick={() => fetchChatMessages(index)}>
+                <Tab.Pane eventKey={index}></Tab.Pane>
+                {patient.name}
+               </Tab.Content>
+        }
+    )
+    return (
+        <div className={styles["overall-container"]}>
+            <div className="select-patient-container">
+            <Tab.Container id="left-tabs-example" defaultActiveKey="-1">
+                {patients.length > 0 ? 
+                <Row>
+                <Col>
+                {patients.map((patient, index) => {
+                return <Nav variant="pills" className="flex-column" onClick={() => fetchChatMessages(index)}>
+                    <Nav.Item>
+                        <Nav.Link eventKey={index}>{patient.name}</Nav.Link>
+                    </Nav.Item>
+               </Nav>
+                })}
+                </Col>
+                <Col sm={9}>
+                {selectedChatIndex === -1 ? 'Select a patient chat' : 
+                <div className={styles["right-side-container"]}>
+                    <div className={styles["chat-box-title"]}>
+                        <div>
+                            <h3><strong>{patients[selectedChatIndex].name}</strong></h3>
+                            {patients[selectedChatIndex].doctorFlagInfo.isFlagged == true 
+                        ? <Badge className={styles["patient-name-display"]} bg={"danger"} >FLAGGED</Badge> 
+                        : null}
+                        </div>
+                        <p>{"PatientID: "+patients[selectedChatIndex].uid}</p>
```

Comment:
Is the PatientID necessary?

Also, it could be written this way
`<p>PatientId: {patients[selectedChatIndex].uid}</p>`